nixos/modules/programs/less.nix

```diff
@@ -0,0 +1,123 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.less;
+
+  configFile = ''
+    #command
+    ${concatStringsSep "\n"
+      (mapAttrsToList (command: action: "${command} ${action}") cfg.commands)
+    }
+    ${if cfg.clearDefaultCommands then "#stop" else ""}
+
+    #line-edit
+    ${concatStringsSep "\n"
+      (mapAttrsToList (command: action: "${command} ${action}") cfg.lineEditingKeys)
+    }
+
+    #env
+    ${concatStringsSep "\n"
+      (mapAttrsToList (variable: values: "${variable}=${values}") cfg.envVariables)
+    }
+  '';
+
+  lessKey = pkgs.runCommand "lesskey"
+            { src = pkgs.writeText "lessconfig" configFile; }
+            "${pkgs.less}/bin/lesskey -o $out $src";
+
+  lessPipe = pkgs.writeScriptBin "lesspipe.sh" ''
+    #! /bin/sh
```

Comment:
Should I change it to `#!${stdenv.shell}`?

Reply:
yes

Reply:
I wonder if [lesspipe](https://github.com/wofr06/lesspipe) would be the better default here. We already package it in nixpkgs.

Reply:
Yup, that's a good idea. It's (almost) always better to use something well maintained than to hack a script together yourself.